### ISSUE TITLE
neofetch: what am I looking at?

### ISSUE THREAD

**Issue by cobalt2727** [chromebrew/chromebrew#7462]
## Description

Getting a bunch of bash errors any time I run it.

If you executed a specific command:
```shell
neofetch --ascii_distro Chrom*
```

Output:
```ruby
bash: _parse_usage: line 16: syntax error near unexpected token `('
bash: _parse_usage: line 16: ` -?(\[)+([a-zA-Z0-9?]))'
bash: error importing function definition for `_parse_usage'
bash: _longopt: line 6: syntax error near unexpected token `('
bash: _longopt: line 6: ` --!(no-*)dir*)'
bash: error importing function definition for `_longopt'
/usr/local/bin/neofetch: line 4783: warning: /home/chronos/user/.config/neofetch/config.conf: warning: script from noexec mount; see https://chromium.googlesource.com/chromiumos/docs/+/master/security/noexec_shell_scripts.md
            .,:loool:,.               chronos@localhost 
        .,coooooooooooooc,.           ----------------- 
     .,lllllllllllllllllllll,.        OS: on Chrome OS x86_64 
    ;ccccccccccccccccccccccccc;       Host: Celes 1.1 
  'ccccccccccccccccccccccccccccc.     Kernel: 4.19.241-13212-ge9b0bab46475 
 ,ooc::::::::okO0000OOkkkkkkkkkkk:    Uptime: 18 days, 2 hours, 2 mins 
.ooool;;;;:xK0kxxxxxk0XK0000000000.   Packages: 271 (crew) 
:oooool;,;OKdddddddddddKX000000000d   Shell: sh 
lllllool;lNdllllllllllldNK000000000   Resolution: 1366x768 
llllllllloMdcccccccccccoWK000000000   WM: chrome 
;cllllllllXXc:::::::::c0X000000000d   Terminal: chrome 
.ccccllllllONkc;,,,;cxKK0000000000.   CPU: Intel Celeron N3050 (2) @ 2.160GHz 
 .cccccclllllxOOOOOOkxO0000000000;    GPU: Intel Atom/Celeron/Pentium Processor x5-E8000/J3xxx/N3xxx 
  .:cccccccclllllllloO0000000OOO,     Memory: 2144MiB / 3885MiB 
    ,:ccccccccclllcd0000OOOOOOl.
      '::cccccccccdOOOOOOOkx:.                                
        ..,::ccccxOOOkkko;.                                   
            ..,:dOkxl:.
```

## Versions and system information

***
Ignoring debug-1.4.0 because its extensions are not built. Try: gem pristine debug --version 1.4.0
- Architecture: `x86_64` (`x86_64`)
- Kernel version: `4.19.241-13212-ge9b0bab46475`

- Chromebrew version: `1.25.4`
- Chromebrew prefix: `/usr/local`
- Chromebrew libdir: `/usr/local/lib64`

- Last update in local repository: cf97ef9 `Git w/o musl, install.sh modified to work w/o git in bootstrap. (#7436) (2 days ago)`

- OS variant: `Chrome OS`
- OS version: `celes-release/R103-14816.131.0`
- OS channel: `stable-channel`
***

## Additional information

oh yeah, what's up with that gem debug error, anyways?

**Comment by satmandu** [comment]
The gem debug error started showing up after a recent glibc fix. Not sure how to fix that yet.

Does a `crew update ; crew upgrade` help?

**Comment by cobalt2727** [comment]
Unfortunately, nope - this has been there for a while now across multiple `crew upgrade`s

**Comment by cobalt2727** [comment]
Just to confirm, still the same behavior after an upgrade

**Comment by satmandu** [comment]
This is an issue with some recent change to bash_completions.

@supechicken Do you want to track this down?

Moving `/usr/local/share/bash-completions` out of the way resolves this issue. (You have to open a new window to see the difference.)

This isn't an issue with bash, I think, since this happens with the built-in bash, so it must be some recent completion file which has broken.

**Comment by cobalt2727** [comment]
I'm not sure if this was a typo on your part, but just to confirm, I don't have a `/usr/local/share/bash-completions` file. Instead I've got a `/usr/local/share/bash-completion/` *directory* with the following inside it:
```bash
chronos@localhost /usr/local/share/bash-completion $ ls
bash_completion  completions  helpers
```

**Comment by satmandu** [comment]
It's definitely an issue of something messing with bash-completions.

I see this bug report from a while ago: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=778282 

So this may be connected to how crosh starts a shell.

In any case, I've opened an issue upstream in case there might be some help in figuring this out: https://github.com/scop/bash-completion/issues/838

**Comment by satmandu** [comment]
@supechicken This is interesting:
When I put this at the top of /usr/local/etc/profile:
```
if [[ "$(coreutils --coreutils-prog=readlink "/proc/$$/exe")" != '/usr/local/bin/bash' ]]; then
  echo "starting chromebrew bash"
  [[ -f  '/usr/local/bin/bash'  ]] && exec /usr/local/bin/bash
else
  coreutils --coreutils-prog=readlink "/proc/$$/exe"
fi
```
and have this in `/usr/local/etc/bash.d/bash`:
```
coreutils --coreutils-prog=readlink "/proc/$$/exe"
if [[ "$(coreutils --coreutils-prog=readlink "/proc/$$/exe")" != '/usr/local/bin/bash' ]]; then
  exec /usr/local/bin/bash
fi
```
I get this from `coreutils --coreutils-prog=readlink "/proc/$$/exe"` twice: `/usr/local/bin/bash`

HOWEVER, when I comment out that section in `/usr/local/etc/profile`
I get this from `coreutils --coreutils-prog=readlink "/proc/$$/exe"`: `/bin/bash`
And then bash gets run again.

It appears that these problems are from invoking shells during the shell startup. We can avoid this by sourcing scripts as much as possible.

In short, a bunch of this mayhem can be handed with this:
Addition to the top of `/usr/local/etc/profile` in the `crew_profile_base` package.
```
if [[ "$(bash -c 'echo $BASH_VERSION')" != "${BASH_VERSION}" ]]; then
  echo "Starting newer Chromebrew bash."
  exec bash
# else
  # echo "Chromebrew bash is not newer"
fi
```
Deletion of `/usr/local/etc/bash.d/bash` in the `bash` package.
Modification of `/usr/local/etc/bash.d/sommelier` in the `sommelier` package thus:
```
source /usr/local/bin/startsommelier
```
Modification of `/usr/local/bin/startsommelier` in the `sommelier` package thus:
Change the `exit 1` line to `return 1`.

Modifications to do that are incoming...

**Comment by satmandu** [comment]
This is working for me now with the above PRs installed.

![image](https://user-images.githubusercontent.com/1096701/198926901-7c6360df-49e9-40c6-b319-08a3584d7b3e.png)
